internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -125,20 +166,46 @@ func (c *controller) Sync(ctx controllerlib.Context) error {
 		}
 	}
 
+	conditions := make([]*metav1.Condition, 0)
+	specCopy := obj.Spec.DeepCopy()
+	var errs []error
+
+	rootCAs, conditions, tlsOk := c.validateTLS(specCopy, conditions)
+	_, conditions, issuerOk := c.validateIssuer(specCopy.Issuer, conditions)
+
+	client := phttp.Default(rootCAs)
+	client.Timeout = 30 * time.Second // copied from Kube OIDC code
+	coreOSCtx := coreosoidc.ClientContext(context.Background(), client)
+
+	pJSON, provider, conditions, providerErr := c.validateProviderDiscovery(coreOSCtx, specCopy, conditions, tlsOk && issuerOk)
+	errs = append(errs, providerErr)
+
+	jwksURL, conditions, jwksErr := c.validateProviderJWKSURL(provider, pJSON, conditions, tlsOk && issuerOk && providerErr == nil)
+	errs = append(errs, jwksErr)
+
 	// Make a deep copy of the spec so we aren't storing pointers to something that the informer cache
-	// may mutate!
-	jwtAuthenticator, err := newJWTAuthenticator(obj.Spec.DeepCopy())
-	if err != nil {
-		return fmt.Errorf("failed to build jwt authenticator: %w", err)
+	// may mutate! We don't store status as status is derived from spec.
+	cachedAuthenticator, conditions, err := c.newCachedJWTAuthenticator(coreOSCtx, client, obj.Spec.DeepCopy(), jwksURL, conditions, tlsOk && issuerOk && providerErr == nil && jwksErr == nil)
+	errs = append(errs, err)
+
+	if !hadErrorCondition(conditions) {
+		c.cache.Store(cacheKey, cachedAuthenticator) // the in-memory cache must need the extra functionality
+		c.log.WithValues("jwtAuthenticator", klog.KObj(obj), "issuer", obj.Spec.Issuer).Info("added new jwt authenticator")
```

Comment:
You can pass keys and values to the `Info` function instead of using `WithValues`. `WithValues` creates a whole new logger.

Reply:
Ah, interesting. I can swap that around.
```
c.log.Info("added new jwt authenticator", "jwtAuthenticator", klog.KObj(obj), "issuer", obj.Spec.Issuer)
```